internal/server/http/server.go

```diff
@@ -48,9 +50,8 @@ func (s *Server) Run(ctx context.Context) error {
 		}
 	}()
 
-	go func() {
-		s.services.SensorService.RunStatusUpdater(ctx, 1*time.Hour)
-	}()
+	scheduler := worker.NewScheduler(ctx, 3*time.Hour, s.services.SensorService)
+	go scheduler.Run()
```

Comment:
Now you have passed the sensor service as a struct and implemented the `Do` function in the service. In this case I would prefer to use the function call. Because now we have a `Do` function inside our service... Perhaps the service has two or more schedulars that will do different things in the future, then this will not work. So please rename the function back to `UpdateStatuses` and passed like so:

```suggestion
	scheduler := worker.NewScheduler(3*time.Hour, worker.SchedulerFunc(s.services.SensorService.UpdateStatuses))
	go scheduler.Run(ctx)
```